service/storage_proxy.cc

```diff
@@ -613,11 +613,14 @@ class storage_proxy::remote {
             co_await coroutine::all(
                 [&] () -> future<> {
                     try {
-                        auto op = _sp.start_write();
                         // FIXME: get_schema_for_write() doesn't timeout
                         schema_ptr s = co_await get_schema_for_write(schema_version, reply_to_host_id, shard, timeout);
                         // Note: blocks due to execution_stage in replica::database::apply()
-                        co_await apply_fn(p, trace_state_ptr, std::move(s), m, timeout, fence);
+                        auto op = p->start_write();
```

Comment:
But why is it safe to move it after getting a schema? What prevent us from writing to already cleaned up range after `get_schema_for_write` returns? Fence?